### PR TITLE
Use compressed CSS

### PR DESCRIPTION
**Pull request by MikeNGarrett**
*TL;DR:* Grunt compiles a minified version, but it doesn't seem to get used, so I added it in. 

As a side note, I am really enjoying this theme. A lot. Thank you for releasing it!
I made a few other modifications you may be interested in: 
- Remove sass flex mixin. Autoprefixer takes care of the -ms prefixes which was good enough for me.
- Change template page-home.php to front-page.php. This may be a preference thing. 
- Add sass globbing. I added a new directory for styles I wanted to load on every page and globbed that directory.
- For future-proofing (not imperative), I changed schema declarations to use https.
- Added escaping functions.

Let me know and I can add PRs for these, too.

### CODE REVIEW

**Comment by joshuaiz** [comment]
Hey @MikeNGarrett thanks for the kind words and thanks for the PR.

I had a look and not entirely sure why you added the WP_DEBUG conditionals. Can you explain the what and the why of those?  Would I be reading it incorrectly to infer that if you did *not* have WP_DEBUG defined and enabled that those styles wouldn't load? WP_DEBUG shouldn't be a requirement.

All that said, I don't use Grunt so if it is Grunt-specific, I'm all for it (I think lol). 

For the other suggestions:
- `page-home.php` vs. `front-page.php` is more a matter of preference. But since WordPress is weird and considers your "front page" the page with your blog roll, we use `page-home.php` for static home pages (which are 99% of the sites we work on). I would leave it as-is but of course you can change it.

- I am woefully unfamiliar with globbing - can you point me to some resources so I can get a gist of what it is? I see the term but I kind of gloss over when I see it. I think it's time to change that :)

- I'm all for the schema using https so yes def send that as a separate PR and I'll merge it right away.

Thanks for this!

**Comment by MikeNGarrett** [comment]
The grunt tasks you have defined write both compressed an uncompressed stylesheets, ie style.min.css, style.map, and style.css. I'm proposing that you use the compressed versions on any site that's not being debugged. 

The condition is written so if WP_DEBUG is defined and it's true, it uses the uncompressed css. All other conditions would use the compressed versions. An alternative would be to use `SCRIPT_DEBUG` since this is used to serve uncompressed WordPress core assets (css and js). 

----------------
I definitely understand your perspective on `front-page.php`. I agree it's a preference, not a requirement. 

globbing is the ability to include everything in a directory, ie `library/scss/modules/*`. Here's some info: https://github.com/DennisBecker/grunt-sass-globbing

I'll submit that PR for schema protocol change. 

Also, I added some escaping to various templates. Most of this is benign, but it's recommended as a best practice for security, eg `echo esc_url( get_theme_file_uri() . '/library/images/icon.svg' );`

**Comment by joshuaiz** [comment]
Makes perfect sense now...thanks. And agree on the escaping - thanks for that. Pulling this in.

**Comment by joshuaiz** [comment]
Regarding sass globbing, I have no problem with that if it only applies to Grunt but on the other hand I don't want to force other Grunt users to use a specific file structure or add on to what's already there.

If this is totally common practice with Grunt then cool but if it is another preferential build step or setup thing, I would say leave things as-is and let people add globbing if they choose.

I have been on the CodeKit train since v1 and it just finds all of the .scss and compiles them to /library/css automatically so it doesn't matter where they are in the directory structure. Thus I am a bit out of the loop with Grunt.

**Comment by MikeNGarrett** [comment]
Understood. I would be 100% onboard with converting this from grunt to CodeKit. It sounds like it's essentially the same thing as what globbing hopes to accomplish.

**Comment by joshuaiz** [comment]
After pulling down this PR one thing I noticed is that the default minimized styles previously in Plate used this format (with a dash): style-min.css instead of how your calls have it: style.min.css.

Lately I've just been using CodeKit to minify scss to css by default since I am only working in the scss files anyway but this brings up another issue - if people don't want to use that syntax or have their default css minified like me, then your WP Debug conditionals won't work.

Ideally, it would be better to have fewer files so I'm going to revert the stylesheet calls back to the previous version. I also removed the extra *-min.css stylesheets that were clogging up the /library/css folder. This way everyone can do it their way.

What I usually do is set CodeKit to output flat when developing and then compressed (minimized) when ready to deploy/go live. This way there is only a single outputted css file (and associated source map) instead of separate expanded and minified files.

I understand what you are trying to do but I don't think it works for everyone and mainly, doesn't work for my setup. Maybe we could come up with a compromise or conditionals based on whether you have Grunt or not or I'm open to any other ideas.

**Comment by MikeNGarrett** [comment]
Gotcha. I understand. The file names were output this way automatically when I first ran grunt, so maybe something changed in a more recent version of cssmin. Using CodeKit in place of an included task runner makes things a bit more difficult to standardize, so I understand your hesitation.
